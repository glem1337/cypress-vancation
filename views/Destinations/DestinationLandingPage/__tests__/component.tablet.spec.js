import React from 'react';
import { shallow } from 'enzyme';

import DestinationsLandingPageTablet from '../component.tablet';
import mockedPageData from '../__mocks__/pageData';

jest.mock('mapbox-gl', () => ({
  Map: jest.fn(() => ({
    on: jest.fn(),
  })),
}));

describe('DestinationsLandingPageTablet component tests', () => {
  const props = {
    pageData: mockedPageData,
    isTabletMapVisible: true,
    toggleTabletMap: jest.fn(),
    createRef: jest.fn(() => jest.fn()),
    isSearchResultsPage: true,
    onDragEnd: jest.fn(),
    onZoomEnd: jest.fn(),
  };

  const component = shallow(<DestinationsLandingPageTablet {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
