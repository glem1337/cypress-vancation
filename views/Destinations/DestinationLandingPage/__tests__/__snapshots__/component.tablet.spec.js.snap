// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationsLandingPageTablet component tests matches snapshot 1`] = `
<Fragment>
  <Connect(InteractiveMapMobile)
    createMapOnMount={true}
    isVisible={true}
    onDragEnd={[MockFunction]}
    onZoomEnd={[MockFunction]}
    toggleTabletMap={[MockFunction]}
  />
  <ViewMapButton
    isVisible={false}
    onClick={[MockFunction]}
  />
  <ContentContainer
    MapSection={null}
    NearbyDestinationsSection={<NearbyDestinations />}
    ResultsSection={
      <div
        className="search-page__results"
      >
        <div
          className="search-page__results-info"
        >
          <SearchResultsInfo
            additionalClassNames="mt-10"
          />
          <EmptyResult />
          <MainPhoto />
          <MainInfo />
          <Memo(Connect(TopLocations)) />
        </div>
        <div
          className="search-page__results-campers"
        >
          <div
            className="search-page__results-text mt-24 mb-24"
          >
            <Description />
            <FanFacts />
          </div>
          <Campers />
        </div>
      </div>
    }
    isDesktopMapVisible={true}
  />
  <Footer
    innerRef={[MockFunction]}
  />
</Fragment>
`;
